src/main/resources/api/openapi.yaml

```diff
@@ -303,6 +315,285 @@ components:
+        digest:
+          $ref: '#/components/schemas/Digest32'
+        treeFlags:
+          type: integer
+          format: int32
+        keyLength:
+          type: integer
+          format: int32
+        valueLength:
+          type: integer
+          format: int32
+          nullable: true
+
+
+    ErgoLikeContext:
+      required:
+       - lastBlockUtxoRoot
+       - headers
+       - preHeader
+       - dataBoxes
+       - boxesToSpend
+       - spendingTransaction
+       - selfIndex
+       - extension
+       - validationSettings
+       - costLimit
+       - initCost
+      properties:
+        lastBlockUtxoRoot:
+          type: object
+          description: state root before current block application
+          $ref: '#/components/schemas/AvlTreeData'
+        headers:
+          type: array
+          description: fixed number of last block headers in descending order (first header is the newest one)
+          items:
+            $ref: '#/components/schemas/SigmaHeader'
+        preHeader:
+          type: object
+          description: fields of block header with the current `spendingTransaction`, that can be predicted by a miner before it's formation
+          $ref: '#/components/schemas/PreHeader'
+        dataBoxes:
+          type: array
+          description: boxes, that corresponds to id's of `spendingTransaction.dataInputs`
+          items:
+            type: ErgoBox
+            $ref: '#/components/schemas/ErgoTransactionOutput'
+        boxesToSpend:
+          type: array
+          description: boxes, that corresponds to id's of `spendingTransaction.inputs`
+          items:
+            type: ErgoBox
+            $ref: '#/components/schemas/ErgoTransactionOutput'
+        spendingTransaction:
+          type: object
+          description: transaction that contains `self` box
+          $ref: '#/components/schemas/ErgoLikeTransaction'
+        selfIndex:
+          type: integer
+          format: int64
+          description: index of the box in `boxesToSpend` that contains the script we're evaluating
+        extension:
+          type: object
+          description: prover-defined key-value pairs, that may be used inside a script
+        validationSettings:
+          type: string
+          description: validataion parameters passed to Interpreter.verify to detect soft-fork conditions
+          example: "10e8070001e9070001ea070001eb070001ec070001ed070001ee070001ef070001f0070001f1070001f2070001f3070001f4070001f5070001f6070001f7070001"
+        costLimit:
+          type: integer
+          format: int64
+          description: hard limit on accumulated execution cost, if exceeded lead to CostLimitException to be thrown
+        initCost:
+          type: integer
+          format: int64
+          description: initial value of execution cost already accumulated before Interpreter.verify is called
+
+
+    ExecuteScript:
+      type: object
+      required:
+      - script
+      - env
+      - ctx
```

Comment:
this should be renamed `ctx`  -> `context`, because it is public API

Reply:
for environment let's keep it short

Reply:
Renamed "ctx" to "context",  but frankly speaking I do not understand logic of such renaming: either we should avoid shortcuts in all cases, either allow to use them. IMHO "ctx" is not more confusing then "env".

Reply:
Ok, let's rename `env` -> `namedConstants`, `ctx` -> `context`.
Environment is misleading, it is almost always confused with script parameters, which is not the case here.

Reply:
Renamed